tests/runtime-tests/src/lib.rs

```diff
@@ -0,0 +1,400 @@
+                manifest_path.display()
+            )
+        })?;
+        let source_str = source.as_str().with_context(|| {
+            format!(
+                "malformed manifest '{}' contains 'source' key that isn't a string in component",
+                manifest_path.display()
+            )
+        })?;
+        // TODO: make this more robust
+        if source_str.starts_with("{{") {
+            let name = &source_str[2..source_str.len() - 2];
+            let path = component_path(components_path, name);
+            let wasm_name = format!("{name}.wasm");
+            std::fs::copy(&path, temp.path().join(&wasm_name)).with_context(|| {
+                format!(
+                    "failed to copy wasm file '{}' to temporary directory",
+                    path.display()
+                )
+            })?;
+            *source = toml::Value::String(wasm_name);
+        }
+    }
+    let manifest =
+        toml::to_string(&table).context("failed to re-serialize manifest as a string")?;
+    std::fs::write(temp.path().join("spin.toml"), manifest)
+        .context("failed to copy spin.toml manifest to temporary directory")?;
+    Ok(())
+}
+
+#[derive(Debug)]
+enum Response {
+    Ok,
+    Err(String),
+    /// This happens when we panic
+    ErrNoBody,
+}
+
+impl FromStr for Response {
+    type Err = anyhow::Error;
+
+    fn from_str(s: &str) -> anyhow::Result<Self> {
+        let code = s
+            .strip_prefix("HTTP/1.1 ")
+            .context("malformed response does not contain `HTTP/1.1` prefix")?[..3]
+            .parse::<u16>()
+            .context("malformed response does not contain a status code")?;
+        anyhow::ensure!(
+            s.as_bytes()[s.len() - 4..] == b"\r\n\r\n"[..],
+            r#"malformed response does not end with the expected CRLF"#
+        );
+        if code == 500 {
+            let header_end = s
+                .find("\r\n\r\n")
+                .context("malformed response does not contain CRLF header separator")?;
+            if header_end + 4 == s.len() {
+                return Ok(Response::ErrNoBody);
+            }
+            let body =
+                String::from_utf8(s.as_bytes()[header_end + 4..s.len() - 4].to_vec()).unwrap();
+            return Ok(Response::Err(body));
+        }
+        if code == 200 {
+            return Ok(Response::Ok);
+        }
+        anyhow::bail!("Could not parse HTTP raw response: {s}")
+    }
+}
+
+fn make_http_request(spin: &mut Spin) -> Result<String, anyhow::Error> {
+    if let Some(status) = spin.try_wait()? {
+        anyhow::bail!("Spin exited early with status code {:?}", status.code());
+    }
+    log::debug!("Connecting to HTTP server on port {}...", spin.port());
+    let mut connection = std::net::TcpStream::connect(format!("127.0.0.1:{}", spin.port()))
+        .context("could not connect to Spin web server")?;
+    connection.write_all(b"GET / HTTP/1.1\r\nHost: 127.0.0.1:3000\r\n\r\n")?;
```

Comment:
will it always be a GET request? and can we make other request types from the tests e.g. to add regression test for https://github.com/fermyon/spin/issues/2080